tsconfig.json

```diff
@@ -1,23 +1,10 @@
 {
+	"extends": "@tsconfig/node14",
 	"compilerOptions": {
-		"incremental": true,
-		"target": "ES2020",
-		"module": "commonjs",
-		"allowJs": false,
+		"outDir": "dist",
+		"composite": true,
 		"sourceMap": true,
-		"declaration": true,
-		"outDir": "./dist",
-		"removeComments": true,
-		"strict": true,
-		"noUnusedLocals": true,
-		"noUnusedParameters": true,
-		"noImplicitReturns": true,
-		"noFallthroughCasesInSwitch": true,
-		"noUncheckedIndexedAccess": true,
```

Comment:
Why have these lines been removed? They are checks that can help us avoid some mistakes and bugs. Is it because of compatibility with the JS libraries?

Reply:
See the commit log for a319808.

Basically, the problem is that these checks are done in the compiler, so they prevent the code from running if they don't pass. This is super annoying because while developing, often I want to quickly comment out a line and then re-run, and not also comment out all the places where the variable was defined/imported.

Most of these checks are already duplicated by our ESLint config (including the unused locals), so you still can't push code that violates them.

The `strict` flag is still enabled by the config we're extending from, see the first line of this file.